src/mumble/Settings.cpp

```diff
@@ -768,7 +770,33 @@ void Settings::load(QSettings *settings_ptr) {
 	SAVELOAD(qsAudioOutput, "audio/output");
 	SAVELOAD(bWhisperFriends, "audio/whisperfriends");
 	SAVELOAD(bTransmitPosition, "audio/postransmit");
-	LOADFLAG(echoOption, "audio/echooptionid");
+
+	if (settings_ptr->contains("audio/echooptionid")) {
+		// Load the new echo cancel option instead
+		LOADFLAG(echoOption, "audio/echooptionid");
+	} else {
+#ifndef Q_OS_MAC
+		// Compatibility layer for overtaking the old (now deprecated) settings
+		// This block should only be called once at the first start of the new Mumble version
+		// As echo cancellation was not available on macOS before, we don't have to run this compatibility
+		// code on macOS (instead simply use the new default as set in the constructor).
+		bool deprecatedEcho      = false;
+		bool deprecatedEchoMulti = false;
```

Comment:
`auto` can be used here without hiding the data type, as we explicitly set the variable's value.

Reply:
But there is also no gain whatsoever in doing so